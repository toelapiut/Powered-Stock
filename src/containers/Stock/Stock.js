import React, {useState, useEffect} from 'react';
import StockScreen from '../../screens/StockScreen';
import useSWR from 'swr';
import {url} from '../../helper/url';
import {toCamelCase} from '../../helper/utils';
import {transposeStock} from '../../helper/utils';
import {quandl} from '../../helper/http';
import PropTypes from 'prop-types';
import {Error} from '../../screens/Error/Error';
import {Loading} from '../../components/Loading/Loading';

const fetcher = url => quandl.post(url).then(res => res.data);

export const Stock = ({ticker}) => {
  const {data, error} = useSWR(`${url.stocks}${ticker}`, fetcher);
  const [start, setStart] = useState(new Date());
  const [end, setEnd] = useState(new Date());
  const [isOpen, setIsOpen] = useState(true);
  const [stocks, setStocks] = useState([]);
  const [details, setDetails] = useState({ticker, name: ''});
  const [loading, setLoading] = useState(true);


  useEffect(() => {
    if (typeof data !== 'undefined') {
      const datasets = toCamelCase(data);
      const {dataset} = datasets;
      const {datasetCode, name} = dataset;
      setDetails({ticker: datasetCode, name});
      const transposed = transposeStock(dataset.data);
      setStocks(transposed);
      setLoading(false);
    }
  }, [data, ticker]);

  useEffect(() => {
    setLoading(false);
  }, [stocks]);

  useEffect(() => {
    setLoading(true);
  }, [ticker]);

  const onChangeDates = (dates) => {
    setStart(dates.start);
    setEnd(dates.end);
    onOpenCalendar();
  };

  const onOpenCalendar = () => {
    setIsOpen(!isOpen);
  };

  if (typeof error !== 'undefined') {
    return <Error/>;
  }

  if (loading) {
    return (
      <div>
        <Loading/>
      </div>
    );
  }
  return (
    <div>
      <StockScreen
        isOpen={isOpen}
        onOpenCalendar={onOpenCalendar}
        name={details.name}
        ticker={details.ticker}
        start={start}
        end={end}
        stocks={stocks}
        onChangeDates={onChangeDates}
      />
    </div>
  );
};

Stock.propTypes = {
  ticker: PropTypes.string
};