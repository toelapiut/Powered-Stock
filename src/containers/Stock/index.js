import {Stock} from './Stock';
import { withToastManager } from 'react-toast-notifications';


export default withToastManager(Stock);