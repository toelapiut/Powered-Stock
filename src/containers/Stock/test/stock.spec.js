import  React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {Stock} from '../Stock';

test('should render correctly', () => {
  const wrapper = shallow(<Stock />);
  expect(toJson(wrapper)).toMatchSnapshot();
});