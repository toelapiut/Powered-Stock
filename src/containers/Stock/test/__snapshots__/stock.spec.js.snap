// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <h1>
    Hello stocks
  </h1>
</div>
`;
